scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -140,6 +144,25 @@
     (write-status-to-datomic kcc status)
     {:expected-state :expected/running}))
 
+(defn record-sandbox-url
+  "Record the sandbox file server URL in datomic."
+  [{:keys [pod]}]
+  (let [task-id (-> pod .getMetadata .getName)
+        pod-ip (-> pod .getStatus .getPodIP)
+        {:keys [default-workdir sandbox-fileserver pod-ip->hostname-fn]} (config/kubernetes)
+        sandbox-fileserver-port (:port sandbox-fileserver)
+        sandbox-url (try
+                      (when (and sandbox-fileserver-port (not (str/blank? pod-ip)))
+                        (str "http://"
+                             (pod-ip->hostname-fn pod-ip)
+                             ":" sandbox-fileserver-port
+                             "/files/read.json?path="
+                             (URLEncoder/encode default-workdir "UTF-8")))
+                      (catch Exception e
+                        (log/debug e "Unable to retrieve directory path for" task-id)
```

Comment:
Why would this throw an exception? Is it being thrown by the `URLEncoder`? Why would that happen?

Like @dposada said, it might be useful to have this logged higher than info-level.